pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -154,29 +154,30 @@ func initializePipelineRunControllerAssets(t *testing.T, d test.Data, opts pipel
 	}, cancel
 }
 
-// getTaskRunCreations will look through a set of actions to find all task run creation actions and return the set of
-// them. It will fatal the test if none are found, or if less actions were found than expected.
-func getTaskRunCreations(t *testing.T, actions []ktesting.Action, minActionCount int) []*v1beta1.TaskRun {
+// getTaskRunCreations will look through a set of taskruns generated by a specific pipelinerun and return the set of
+// them. It will fatal the test if none are found or an error occurred
+func getTaskRunCreations(ctx context.Context, t *testing.T, clients test.Clients, namespace string, prName string) []*v1beta1.TaskRun {
 	t.Helper()
-	if len(actions) < minActionCount {
-		t.Fatalf("Expected client to have at least %d action implementation but it has %d", minActionCount, len(actions))
+
+	opt := metav1.ListOptions{
+		LabelSelector: pipeline.PipelineRunLabelKey + "=" + prName,
+	}
+
+	taskRuns, err := clients.Pipeline.TektonV1beta1().TaskRuns(namespace).List(ctx, opt)
+	if err != nil {
+		t.Fatalf("failed to list taskruns, %s", err)
 	}
 
 	var outputs []*v1beta1.TaskRun
-	for _, a := range actions {
-		if action, ok := a.(ktesting.CreateAction); ok {
-			if output, ok := action.GetObject().(*v1beta1.TaskRun); ok {
-				outputs = append(outputs, output)
-			}
-		}
+	for _, item := range taskRuns.Items {
+		tr := item
```

Comment:
NIT: we might fetch tr in the iterations to save one line:)
ie. for _, tr := range taskRuns.Items

Reply:
Thanks for the feedback, &tr is added to the outputs slice, If the tr variable is modified after the iteration is complete, the values in the outputs slice may also change.

Reply:
> NIT: we might fetch tr in the iterations to save one line:) ie. for _, tr := range taskRuns.Items

Thanks for the feedback, &tr is added to the outputs slice, If the tr variable is modified after the iteration is complete, the values in the outputs slice may also change.